src/catalog/src/durable/impls/persist.rs

```diff
@@ -508,10 +616,12 @@ impl PersistCatalogState {
                         apply(&mut self.snapshot.default_privileges, key, value, diff);
                     }
                     StateUpdateKind::Epoch(epoch) => {
-                        return Err(DurableCatalogError::Fence(format!(
-                            "current catalog epoch {} fenced by new catalog epoch {}",
-                            self.epoch, epoch
-                        )));
+                        if diff == 1 && epoch != self.epoch {
```

Comment:
Why the `diff == 1` check? Feels like this could use some commentary and I'd also be more explicit about whichever diff values we are ignoring (e.g. if we're ignoring `diff == 0`, then have a chain of if/else where we might explicitly do nothing for `diff == 0` and then panic if it's an entirely unexpected value)

Reply:
At the beginning of this match branch there's a match guard `if diff == 1 || diff == -1`, otherwise we panic and print an error message about an invalid update.

We want to ignore `diff == -1` because we'll end up seeing that when the previous epoch is retracted. I've added a comment explaining this.

Reply:
Ah, whoops, sorry, github was hiding that from me and I didn't know to expand the diff. Thanks!